### PR TITLE
[FAB-17582] Use custom API type for FabricMSPConfig proto

### PR DESCRIPTION
**Pull request by caod123**
#### Type of change

- Improvement (improvement to code, performance, etc)
- Test update

#### Description

- Custom API type is more consumer friendly. It accepts x509
certificates and ECDSA keys instead of byte slices and handles PEM
encoding them to byte slices for the FabricMSPConfig proto.
- General cleanup for application channel creation.
- Cleanup for AddOrgToConsortium

#### Related issues

[FAB-17582](https://jira.hyperledger.org/browse/FAB-17582)

### CODE REVIEW

File: pkg/config/config_test.go
Comment (caod123) [reply pkg/config/config_test.go]: We figured out this was because we were using protolator on an arbitrary ConfigGroup. protolator is unable to determine how to decorate the ConfigGroup if we don't explicitly say what type of ConfigGroup it is. By wrapping it with a DynamicOrdererOrgGroup in the tests it now properly decodes the values.

File: pkg/config/msp.go
Comment (mastersingh24) [inline pkg/config/msp.go]:
```suggestion
func buildPemEncodedCRL(crls []pkix.CertificateList) ([][]byte, error) {
```

File: pkg/config/msp.go
Comment (mastersingh24) [inline pkg/config/msp.go]:
```suggestion
	for _, crl := range crls {
```

File: pkg/config/msp.go
Comment (mastersingh24) [inline pkg/config/msp.go]:
```suggestion
		asn1MarshalledBytes, err := asn1.Marshal(crl)
```

File: pkg/config/msp.go
Comment (mastersingh24) [reply pkg/config/msp.go]: mostly a nit, but the input is really an array of crls

**Review by mastersingh24 — CHANGES_REQUESTED** [review]

**Review by mastersingh24 — DISMISSED** [review]

**Comment by mastersingh24** [comment]
I'll let @sykesm have one more look as well

File: pkg/config/application.go
Comment (sykesm) [inline pkg/config/application.go]: It's not specific to this function (and I think I've mentioned it before), please consider value objects on the API.

File: pkg/config/config.go
Comment (sykesm) [inline pkg/config/config.go]: Should these be references or values?

File: pkg/config/msp.go
Comment (sykesm) [inline pkg/config/msp.go]: values or references? In the case of the slices, they are references but should they contain pointers or values?

**Review by sykesm — DISMISSED** [review]
This can be merged if you're comfortable there won't be a broken build due to a conflict.

File: pkg/config/application.go
Comment (caod123) [reply pkg/config/application.go]: Good point I'll push a quick followon commit to this PR that makes some changes to the types.

**Comment by caod123** [comment]
@sykesm we pushed a followon commit to this PR that addresses some of the unnecessary pointers in the config package. This PR was also rebased on master so if CI passes it shouldn't break master.

File: pkg/config/application.go
Comment (sykesm) [inline pkg/config/application.go:22]: Pointer or value?

File: pkg/config/application.go
Comment (sykesm) [inline pkg/config/application.go:147]: Given all the other functions take a pointer, this will be inconsistent and require a dereference.

**Review by sykesm — APPROVED** [review]

File: pkg/config/application.go
Comment (DereckLuo) [reply pkg/config/application.go:22]: it is easier to use it as pointers, as for now many tests are modifying policy values, which using pointer make it a bit easier. We can consider changing it to value down the line.

**Pull request merged**
